### PR TITLE
New function mbedtls_pk_copy_public_from_psa

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Follow-up to https://github.com/Mbed-TLS/mbedtls/pull/8774#discussion_r1513205206: during the review, I identified a design gap: we're missing a convenient way to export the public key of a PSA key into PK.

Status: targeting the merge queue of #8774. The commits should be good to go.

## PR checklist

- [x] **changelog** provided
- [x] **backport** no (new feature)
- [x] **tests** provided
- [x] Update https://github.com/Mbed-TLS/mbedtls/pull/8889 to mention `mbedtls_pk_copy_public_from_psa`

### CODE REVIEW

File: tests/suites/test_suite_pk.function [hunk 1]
```diff
@@ -2404,9 +2488,11 @@ void pk_copy_from_psa_success(data_t *priv_key_data, int key_type_arg,
                                  key_type, key_usage, key_alg, &priv_key_id));
     pub_key_id = psa_pub_key_from_priv(priv_key_id);
 
-    /* Create 2 PK contexts starting from the PSA keys we just created. */
+    /* Create 4 PK contexts starting from the PSA keys we just created. */
     TEST_EQUAL(mbedtls_pk_copy_from_psa(priv_key_id, &pk_priv), 0);
+    TEST_EQUAL(mbedtls_pk_copy_public_from_psa(pub_key_id, &pk_priv_copy_public), 0);
```
Comment (valeriosetti) [inline tests/suites/test_suite_pk.function:2493]: I think it's:
```suggestion
    TEST_EQUAL(mbedtls_pk_copy_public_from_psa(priv_key_id, &pk_priv_copy_public), 0);
```

**Review by valeriosetti — COMMENTED** [review]
I only found 1 fix, but a part from that this looks OK to me. I will approve once it will be rebased on #8774

**Review by mpg — CHANGES_REQUESTED** [review]
Looks good to me other than the issue already pointed out by Valerio.

**Review by valeriosetti — APPROVED** [review]
LGTM :)

**Review by mpg — DISMISSED** [review]
LGTM

**Comment by gilles-peskine-arm** [comment]
I changed the base branch now that the preceding PR is merged. Please re-approve.

**Review by mpg — APPROVED** [review]

**Review by valeriosetti — APPROVED** [review]

**Pull request merged**